Source/CesiumRuntime/Private/CesiumGltfPrimitiveComponent.cpp

```diff
@@ -183,14 +183,10 @@ struct CalcBoundsOperation {
         glm::max(glm::length(halfAxes[0]), glm::length(halfAxes[1]));
     sphereRadius = glm::max(sphereRadius, glm::length(halfAxes[2]));
 
-    FVector xs(halfAxes[0].x, halfAxes[1].x, halfAxes[2].x);
-    FVector ys(halfAxes[0].y, halfAxes[1].y, halfAxes[2].y);
-    FVector zs(halfAxes[0].z, halfAxes[1].z, halfAxes[2].z);
-
     FBoxSphereBounds result;
     result.Origin = VecMath::createVector(center);
     result.SphereRadius = sphereRadius;
-    result.BoxExtent = FVector(xs.GetAbsMax(), ys.GetAbsMax(), zs.GetAbsMax());
+    result.BoxExtent = FVector(sphereRadius, sphereRadius, sphereRadius);
```

Comment:
There's still a comment that says the sphere only has to reach the side of the box, is that still applicable / correct?

Also is all the stuff with the half-axes necessary in the BoundingSphere conversion, can't we just convert the sphere radius from meters to centimeters? Could we assume getTilesetToUnrealWorld is a rigid transform + a meters to centimeters scaling?

Reply:
I was trying to avoid making any assumptions about the transformation, since it's just a parameter to the method. So instead I construct a bounding box around the sphere, transform the box, and then build a sphere inset in the box. The comment is referring to the fact that the sphere only needs to be inset in the box, rather than fully contain it. There could very well be a better way to do this, but I think it works reasonably well.